src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs

```diff
@@ -125,7 +125,7 @@ internal XmlResolver GetXmlResolver()
         //notice we must keep GetXmlResolver() to avoid dead lock when init System.Config.ConfigurationManager
         internal XmlResolver GetXmlResolver_CheckConfig()
         {
-            if (!IsXmlResolverSet)
+            if (!LocalAppContextSwitches.DontProhibitDefaultResolver && !IsXmlResolverSet)
```

Comment:
@danmosemsft I believe there was an unwritten rule with `AppContextSwitches` that the default value needs to be false and that's why we initially named the switch this way. @AlexGhiondea probably has more context on this though.